src/Control/Distributed/Process/Internal/Messaging.hs

```diff
@@ -135,24 +137,28 @@ connBetween node from to implicitReconnect = do
         setupConnBetween node from to implicitReconnect
 
 disconnect :: LocalNode -> Identifier -> Identifier -> IO ()
-disconnect node from to =
-  modifyValidLocalState_ node $ \vst ->
+disconnect node from to = do
```

Comment:
> We want the connection to be closed eventually, but masking assumes that NT.close is uninterruptible or it won't accomplish anything. And I don't think NT.close is uninterruptible.

I'd like n-t semantics to say smth here, however I agree with the point.
I think `mask_ $ do { ... ; forkIO $ forM_ mio id }`, like you are suggesting will be a sane solution.